libraries/AP_Frsky_Telem/AP_Frsky_Telem.h

```diff
@@ -152,11 +83,20 @@ class AP_Frsky_Telem : public AP_RCTelemetry {
         HOME =          7,  // 0x5004 Home
         BATT_2 =        8,  // 0x5008 Battery 2 status
         BATT_1 =        9,  // 0x5008 Battery 1 status
-        PARAM =         10  // 0x5007 parameters
+        PARAM =         10, // 0x5007 parameters
+#if HAL_WITH_FRSKY_TELEM_BIDIRECTIONAL
+        MAV =           11  // 0x5007 parameters
+#endif //HAL_WITH_FRSKY_TELEM_BIDIRECTIONAL
     };
 
-    struct
-    {
+#if HAL_WITH_FRSKY_TELEM_BIDIRECTIONAL
+    typedef struct {
```

Comment:
We now have `ObjectBuffer_TS` which is... pretty much exactly this.  You'll be able to cut out a lot of `WITH_SEM` calls when you switch to it.

Reply:
ok thanks, migrated to ObjectBuffer_TS